bug recreating a fixed array when calling a direct function with limited len argument array to output(stdio)

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** 
```
V 0.2.4 727c9fb
```
**OS:** 
```
Ubuntu 20.04

Linux 5.11.22-5-pve #1 SMP PVE 5.11.22-10 x86_64 x86_64 x86_64 GNU/Linux
```
**What did you do?**

```
fn main() {
        test(([1]!))
}

fn test(t [1]int) {
        println(t)
}
```
**What did you expect to see?**
```
' [1]'
```
**What did you see instead?**

```
==================
/tmp/v_0/test.18086548121711758600.tmp.c:11078: error: ';' expected (got "}")
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -78,6 +78,16 @@ fn (mut g Gen) array_init(node ast.ArrayInit) {
 			g.inside_lambda = false
 			return
 		}
+		need_tmp_var := g.inside_call && !g.inside_struct_init
+		mut stmt_str := ''
+		mut tmp_var := ''
+		if need_tmp_var {
+			tmp_var = g.new_tmp_var()
+			stmt_str = g.go_before_stmt(0)
+			ret_typ := g.typ(node.typ)
+			g.empty_line = true
+			g.write('$ret_typ $tmp_var = ')
+		}
 		g.write('{')
 		if node.has_val {
 			for i, expr in node.exprs {
@@ -100,6 +110,11 @@ fn (mut g Gen) array_init(node ast.ArrayInit) {
 			g.write('0')
 		}
 		g.write('}')
+		if need_tmp_var {
+			g.writeln(';')
+			g.write(stmt_str)
+			g.write(tmp_var)
+		}
 		return
 	}
 	elem_styp := g.typ(elem_type.typ)

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -113,6 +113,7 @@ mut:
 	inside_match_optional  bool
 	inside_vweb_tmpl       bool
 	inside_return          bool
+	inside_struct_init     bool
 	inside_or_block        bool
 	inside_call            bool
 	inside_for_c_stmt      bool
@@ -3390,7 +3391,9 @@ fn (mut g Gen) expr(node ast.Expr) {
 				g.expr(ast.resolve_init(node, g.unwrap_generic(node.typ), g.table))
 			} else {
 				// `user := User{name: 'Bob'}`
+				g.inside_struct_init = true
 				g.struct_init(node)
+				g.inside_struct_init = false
 			}
 		}
 		ast.TypeNode {

--- a/vlib/v/tests/fn_call_fixed_array_literal_args_test.v
+++ b/vlib/v/tests/fn_call_fixed_array_literal_args_test.v
@@ -1,0 +1,9 @@
+fn test_fn_call_fixed_array_literal_args() {
+	ret := get_str([1]!)
+	assert ret == '[1]'
+}
+
+fn get_str(t [1]int) string {
+	println(t)
+	return '$t'
+}

